research/python/src/data/provider.py

```
@@ -0,0 +1,1077 @@
+                    if futures_contracts:
+                        contract = futures_contracts[0]
+
+                        futures_price = float(contract.get("lastPrice", 0))
+                        oi = int(contract.get("openInterest", 0))
+                        volume = int(contract.get("totalTradedVolume", 0))
+
+                        # Parse expiry date
+                        expiry_str = contract.get("expiryDate", "")
+                        try:
+                            expiry_date = datetime.strptime(expiry_str, "%d-%b-%Y")
+                        except:
```

Ruff (0.14.10)

Do not use bare `except`

(E722)